src/effects/effectslot.cpp

```diff
@@ -268,3 +272,58 @@ void EffectSlot::slotEffectMetaParameter(double v, bool force) {
         pParameterSlot->onEffectMetaParameterChanged(v, force);
     }
 }
+
+QDomElement EffectSlot::toXML(QDomDocument* doc) const {
+    QDomElement effectElement = doc->createElement(EffectXml::Effect);
+    if (m_pEffect == nullptr) {
+        return effectElement;
+    }
+
+    QDomElement metaKnobElement = doc->createElement(EffectXml::EffectMetaParameter);
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectMetaParameter,
+                         QString::number(m_pControlMetaParameter->get()));
+    EffectManifest manifest = m_pEffect->getManifest();
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectId, manifest.id());
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectVersion, manifest.version());
+
+    QDomElement parametersElement = doc->createElement(EffectXml::ParametersRoot);
+
+    for (const auto& pParameter : m_parametersById) {
+        parametersElement.appendChild(pParameter->toXML(doc));
+    }
+
+    effectElement.appendChild(parametersElement);
+
+    return effectElement;
+}
+
+void EffectSlot::loadValuesFromXml(const QDomElement& effectElement) {
```

Comment:
We have some loadValuesFromXml() function all do different things. Can we name this accordingly? 
This could become loadParametersFromXml() 
And below: 
pParameterSlot->loadValueFromXml()